base/ca/src/com/netscape/ca/CAService.java

```diff
@@ -1206,44 +1239,42 @@ Extension createSCTextension(List<String> ctResponses) {
-        byte ct_signature[] = CryptoUtil.base64Decode(response.getSignature());
-        logger.debug(method + " ct_signature len = "+ ct_signature.length);
-        // hard code for now for SHA256withEC signature size is 64
-        int sig_index = ct_signature.length - 64;
-        /*
-         * first 4 bytes are for algorithms
-         * e.g. 04030047 (sha256, ecdsa)
-         */
-        byte[] signature = Arrays.copyOfRange(ct_signature, sig_index, ct_signature.length);
-        logger.debug(method + " signature len = "+ signature.length);
+        try {
+            long timestamp_l = response.getTimestamp();
+            TimeStamp timestamp_t = new TimeStamp(timestamp_l);
+            String timestamp_s = timestamp_t.toString();
+            logger.debug(method + " ct_timestamp: " + timestamp_s);
+            // timestamp
+            byte timestamp[] = timeStampHexStringToByteArray(timestamp_s);
+            byte ct_signature[] = CryptoUtil.base64Decode(response.getSignature());
+            logger.debug(method + " ct_signature len = "+ ct_signature.length);
+            // hard code for now for SHA256withEC signature size is 64
+            int sig_index = ct_signature.length - 64;
+            /*
+             * first 4 bytes are for algorithms
+             * e.g. 04030047 (sha256, ecdsa)
+             */
+            byte[] signature = Arrays.copyOfRange(ct_signature, sig_index, ct_signature.length);
+            logger.debug(method + " signature len = "+ signature.length);
 
-        /* compose data */
-        byte[] version = new byte[] {0}; // v1(0)
-        byte[] signature_type = new byte[] {0}; // certificate_timestamp(0)
-        byte[] entry_type = new byte[] {1}; // LogEntryType: precert_entry(1)
+            /* compose data */
+            byte[] version = new byte[] {0}; // 1 byte; v1(0)
+            byte[] signature_type = new byte[] {0}; // 1 byte; certificate_timestamp(0)
+            byte[] entry_type = new byte[] {0, 1}; // 2 bytes; LogEntryType: precert_entry(1)
 
-        logger.debug(method + "using CT log public key: " + logPublicKey);
-        byte logPublicKey_b[] = CryptoUtil.base64Decode(logPublicKey);
+            logger.debug(method + "using CT log public key: " + logPublicKey);
+            byte logPublicKey_b[] = CryptoUtil.base64Decode(logPublicKey);
 
-        // First, verify the log id
-        PublicKey log_pubKey = KeyFactory.getInstance("EC", "Mozilla-JSS").generatePublic(
-                new X509EncodedKeySpec(logPublicKey_b));
+            // First, verify the log id
+            PublicKey log_pubKey = KeyFactory.getInstance("EC", "Mozilla-JSS").generatePublic(
+                    new X509EncodedKeySpec(logPublicKey_b));
 
-        byte[] log_key_hash = null;
-        try {
+            byte[] log_key_hash = null;
             MessageDigest SHA256Digest = MessageDigest.getInstance("SHA256");
 
             log_key_hash = SHA256Digest.digest(log_pubKey.getEncoded());
```

Comment:
Does `getEncoded()` return a DER-encoded SubjectPublicKeyInfo?  That is what it needs to be.
If it is not, maybe digest `logPublicKey_b` which should be in the correct format.